src/main/java/org/ibp/api/rest/germplasm/GermplasmNameResource.java

```diff
@@ -33,6 +38,24 @@
 		return new ResponseEntity<>(this.germplasmNameService.createName(programUUID, germplasmNameRequestDto, gid), HttpStatus.OK);
 	}
 
+	@ApiOperation(value = "Create code name (CODE1, CODE2, CODE2) for specified list of germplasm")
+	@PreAuthorize("hasAnyAuthority('ADMIN', 'GERMPLASM', 'MANAGE_GERMPLASM', 'CODE_GERMPLASM')")
+	@RequestMapping(value = "/crops/{cropName}/germplasm/coding", method = RequestMethod.POST)
+	@ResponseBody
+	public ResponseEntity<List<GermplasmCodingResult>> createGermplasmCodeNames(@PathVariable final String cropName,
+		@RequestBody final GermplasmCodeNameBatchRequestDto germplasmCodeNameBatchRequestDto) {
+		return new ResponseEntity<>(this.germplasmNameService.createCodeNames(germplasmCodeNameBatchRequestDto), HttpStatus.OK);
+	}
+
+	@ApiOperation(value = "Get next code name sequence based on the specified name settings")
+	@PreAuthorize("hasAnyAuthority('ADMIN', 'GERMPLASM', 'MANAGE_GERMPLASM', 'CODE_GERMPLASM')")
+	@RequestMapping(value = "/crops/{cropName}/germplasm/names/next-name-generation", method = RequestMethod.POST)
```

Comment:
next-generation should be enough because we are already referring to names in the URL

Reply:
done